// openclaw/dist/slash-dispatch.runtime-DNjc4obN.js
import { t as resolveConversationLabel$1 } from "./conversation-label-BIIvASp6.js";
import { s as resolveChunkMode$1 } from "./chunk-BbwYgLMX.js";
import { i as resolveAgentRoute$1 } from "./resolve-route-BNuWCNW8.js";
import { t as finalizeInboundContext$1 } from "./inbound-context-CGEt-zdB.js";
import { n as dispatchReplyWithDispatcher$1 } from "./provider-dispatcher-DCbApMuG.js";
import "./routing-D5xKv1Mn.js";
import { t as resolveMarkdownTableMode$1 } from "./markdown-tables-CtsSO9pl.js";
import "./reply-runtime-Ke_lK6wQ.js";
import { t as recordInboundSessionMetaSafe$1 } from "./conversation-runtime-qvy2dr_3.js";
import "./markdown-table-runtime-VN9LlLeQ.js";
import { r as deliverSlackSlashReplies$1 } from "./replies-BaPVhTtU.js";
//#region extensions/slack/src/monitor/slash-dispatch.runtime.ts
function resolveChunkMode(...args) {
	return resolveChunkMode$1(...args);
}
function finalizeInboundContext(...args) {
	return finalizeInboundContext$1(...args);
}
function dispatchReplyWithDispatcher(...args) {
	return dispatchReplyWithDispatcher$1(...args);
}
function resolveConversationLabel(...args) {
	return resolveConversationLabel$1(...args);
}
function recordInboundSessionMetaSafe(...args) {
	return recordInboundSessionMetaSafe$1(...args);
}
function resolveMarkdownTableMode(...args) {
	return resolveMarkdownTableMode$1(...args);
}
function resolveAgentRoute(...args) {
	return resolveAgentRoute$1(...args);
}
function deliverSlackSlashReplies(...args) {
	return deliverSlackSlashReplies$1(...args);
}
//#endregion
export { deliverSlackSlashReplies, dispatchReplyWithDispatcher, finalizeInboundContext, recordInboundSessionMetaSafe, resolveAgentRoute, resolveChunkMode, resolveConversationLabel, resolveMarkdownTableMode };
